api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -856,14 +858,16 @@ private void stopOrder(Order orderToStop, Date discontinueDate, boolean isRetros
 		if (DISCONTINUE == orderToStop.getAction()) {
 			throw new CannotStopDiscontinuationOrderException();
 		}
-		
-		if (isRetrospective && orderToStop.getDateStopped() != null) {
-			throw new CannotStopInactiveOrderException();
-		}
-		if (!isRetrospective && !orderToStop.isActive()) {
-			throw new CannotStopInactiveOrderException();
-		} else if (isRetrospective && !orderToStop.isActive(discontinueDate)) {
-			throw new CannotStopInactiveOrderException();
+
+		if (!Boolean.parseBoolean(ConfigUtil.getProperty(OpenmrsConstants.GP_ALLOW_SETTING_STOP_DATE_ON_INACTIVE_ORDERS, "false"))) {
```

Comment:
Wouldn't be the worst thing to add a `ConfigUtil.getPropertyAsBoolean(String property, boolean default)` method in, which you can use here and in the setting below...